Tests/CareKitEssentialsTests/CareTaskProgressStrategynTests.swift

```diff
@@ -0,0 +1,535 @@
+        let outcomeValues = [
+         OCKOutcomeValue(10),
+         OCKOutcomeValue(15),
+         OCKOutcomeValue(20),
+         OCKOutcomeValue(30),
+         OCKOutcomeValue(40)
+        ]
+        let targetValues = [
+        OCKOutcomeValue(15),
+        OCKOutcomeValue(25),
+        OCKOutcomeValue(35)
+        ]
+        let event = OCKAnyEvent.mock(
+            taskUUID: UUID(),
+            occurrence: 0,
+            hasOutcome: true,
+            values: outcomeValues,
+            targetValues: targetValues
+        )
+        let progress = CareTaskProgressStrategy<LinearCareTaskProgress>
+            .computeProgressByAveragingOutcomeValues(for: event)
+        let expectedValue = outcomeValues
+            .map { $0.numberValue?.doubleValue ?? 0.0 }
+            .reduce(0, +) / Double(outcomeValues.count)
+        let expectedGoal = targetValues
+            .map { $0.numberValue?.doubleValue ?? 0.0 }
+            .reduce(0, +)
+            XCTAssertEqual(progress.value, expectedValue, accuracy: 0.0001)
+            XCTAssertEqual(progress.goal!, expectedGoal, accuracy: 0.0001)
+    }
+
+    func testProgressByAveragingOutcomeValuesZeroOutcomeTargetValues() async throws {
+        let targetValues = [
+            OCKOutcomeValue(15.0),
+            OCKOutcomeValue(25.0),
+            OCKOutcomeValue(35.0)
+        ]
+        let event = OCKAnyEvent.mock(
+            taskUUID: UUID(),
+            occurrence: 0,
+            hasOutcome: false,
+            targetValues: targetValues
+        )
+        let progress =
+        CareTaskProgressStrategy<LinearCareTaskProgress>
+            .computeProgressByAveragingOutcomeValues(for: event)
+            XCTAssertEqual(progress.value, 0.0, accuracy: 0.0001)
+            let expectedGoal = targetValues
+            .map { $0.doubleValue ?? 0.0 }
+            .reduce(0, +)
+            XCTAssertEqual(progress.goal!, expectedGoal, accuracy: 0.0001)
+    }
+
+    func testProgressByAveragingOutcomeValuesOutcomeValuesNoTargetValues() async throws {
+        let outcomeValues = [
+            OCKOutcomeValue(10.0),
+            OCKOutcomeValue(20.0),
+            OCKOutcomeValue(15.0)
+        ]
+        let event = OCKAnyEvent.mock(
+            taskUUID: UUID(),
+            occurrence: 0,
+            hasOutcome: true,
+            values: outcomeValues
+        )
+        let progress = CareTaskProgressStrategy<LinearCareTaskProgress>
+            .computeProgressByAveragingOutcomeValues(for: event)
+        let expectedValue = outcomeValues
+            .map { $0.doubleValue ?? 0.0 }
+            .reduce(0, +) / Double(outcomeValues.count)
+        XCTAssertEqual(progress.value, expectedValue, accuracy: 0.0001)
+        XCTAssertNil(progress.goal)
+    }
+
+    func testProgressByAveragingOutcomeValuesWithMatchingKind() async throws {
+        var valueOfTen = OCKOutcomeValue(10.0)
+        valueOfTen.kind = "myKind"
```

Comment:
We want to reduce the possibility of programatic mistakes. Can you create two `String` variables at the top of the method and use them throughout?:

```swift
let kind = "myKind"
let kind2 = "otherKind"

var valueOfTen = OCKOutcomeValue(10.0)
valueOfTen.kind = kind
.
.
.
var valueOfThirty = OCKOutcomeValue(30.0)
valueOfThirty.kind = kind2
```